### PR TITLE
Adding in CONTRIBUTING and CODINGSTANDARDS

### PR DESCRIPTION
**Pull request by dalefrancis88**
The purpose of this PR is to close out this issue, https://github.com/evilsoft/crocks/issues/88

### CODE REVIEW

**Comment by evilsoft** [comment]
So I ❤️ the direction this is going!
One thing though to keep diffs and changes easier to manage, lets all a 80 char limit on a line for these large paragraphs

**Comment by dalefrancis88** [comment]
Cool sounds good, wasn't sure wether to keep that rule for these or not. Good to know

**Comment by evilsoft** [comment]
yep, maybe should add it to the styles 😸

**Comment by dalefrancis88** [comment]
@evilsoft @bennypowers Because i used `find` as the example, i've updated it to match the standards i was using it to represent :)

File: CODINGSTANDARDS.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
+# Coding conventions for crocks
+
+## tl;dr;
+
+* `npm run setup` installs all the things
+* `npm run lint` to check for styling issues
+* `npm run spec:dev` to have tests running while coding
+* `npm run spec:coverage` to ensure coverage is within expected levels
```
Comment (evilsoft) [inline CODINGSTANDARDS.md]: you may wanna add `npm test` to run all the things that would be run on CI

File: CODINGSTANDARDS.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+the details of the author within the structure of the following license statement
+
+```javascript
+/** @license ISC License (c) copyright 2018 original and current authors */
+/** @author Dale Francis (dalefrancis88) */
+```
+
+#### Imports
+
+The imports should be in two alphabetized groups with monoids as the first
```
Comment (evilsoft) [inline CODINGSTANDARDS.md]: `monoids` -> data types

File: CODINGSTANDARDS.md [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+group and functions as the second.
+
+```javascript
+const Pred = require('../core/types').proxy('Pred')
+
+const curry = require('../core/curry')
+const predOrFunc = require('../core/predOrFunc')
```
Comment (evilsoft) [inline CODINGSTANDARDS.md]: This is not alphabetized

File: CODINGSTANDARDS.md [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+const isFunction = require('../core/isFunction')
+const isFoldable = require('../core/isFoldable')
```
Comment (evilsoft) [inline CODINGSTANDARDS.md]: Should be above `isFunction`

File: CODINGSTANDARDS.md [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+const isSameType = require('../core/isSameType')
+```
+
+#### Private functions
```
Comment (evilsoft) [inline CODINGSTANDARDS.md]: `Private` -> `Local`

File: CODINGSTANDARDS.md [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+
+Private functions or destructing functions from your imports are done between
```
Comment (evilsoft) [inline CODINGSTANDARDS.md]: `Private functions or destructing functions` -> `Local or destructured functions`

File: CODINGSTANDARDS.md [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+```javascript
+const { Just, Nothing } = require('.')
+
+const accumulator = fn => (acc, cur) =>
+  !acc.found && predOrFunc(fn, cur) ? { found: true, value: cur } : acc
+```
+
+#### Main function
+
+The main function must have a function signature written above using the haskell
```
Comment (evilsoft) [inline CODINGSTANDARDS.md]: `haskell arrow notation` -> `Hindley–Milner type notation`

File: CODINGSTANDARDS.md [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+arrow notation. If you need help, just ask on the [Gitter][gitter] channel.
+Each argument should be validated to be it's expected type, throwing 
+a `TypeError` if it is not. Errors should use a consistent voice, preferring
```
Comment (evilsoft) [inline CODINGSTANDARDS.md]: `Errors` -> `Error messages`

File: CODINGSTANDARDS.md [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+
+```javascript
+// find :: Foldable f => ((a -> Boolean) | Pred) -> f a -> Maybe a
+function find(fn, foldable) {
+  if(!isFunction(fn) && !isSameType(Pred, fn)) {
+    throw new TypeError('find: First argument must be a Pred or predicate')
+  }
+
+  if(!isFoldable(foldable)) {
+    throw new TypeError('find: Second argument must be a foldable')
```
Comment (evilsoft) [inline CODINGSTANDARDS.md]: `foldable` -> `Foldable`

File: CODINGSTANDARDS.md [hunk 10]
```diff
@@ -0,0 +1,241 @@
...
+  t.throws(fn({}), err, 'throws if fn is empty POJO')
+  t.throws(fn({ hi: 'there' }), err, 'throws if fn is non-empty POJO')
+
+  t.end()
+})
+```
+
+## Documentation Standards
+
+If you become familiar with the documentation, it's structure and format will
```
Comment (evilsoft) [inline CODINGSTANDARDS.md]: `it's` -> `its`

File: CODINGSTANDARDS.md (hunk 7 above)
Comment (dalefrancis88) [reply CODINGSTANDARDS.md]: THATS THE NAME! omg i searched everywhere for that and all i kept getting was `haskell notation`

**Comment by dalefrancis88** [comment]
np thanks for reviewing :) Next will be the help file upgrade and then mono-repo

**Pull request merged**